person/src/homeworks/person/core/model/BasePerson.java

```diff
@@ -1,55 +1,110 @@
 package homeworks.person.core.model;
 
+import homeworks.person.core.exceptions.InvalidAgeException;
+import homeworks.person.core.exceptions.InvalidGenderException;
+import homeworks.person.core.exceptions.InvalidNameException;
 import homeworks.person.core.service.PersonAction;
+import homeworks.person.utils.Validate;
 
-public class BasePerson implements PersonAction {
+public abstract class BasePerson implements PersonAction {
     protected String name;
-    protected String designation;
+    protected String lastName;
+    protected String surname;
+    protected String nickname;
+    protected int age;
+    protected String email;
+    protected String  gender;
 
-    public BasePerson(String name, String designation)
+    protected PersonAction actions;
+
+    public BasePerson() {}
+
+
+    public void setName(String name)
     {
-        this.name = name;
-        this.designation = designation;
-    }
-    /**
-     * Set the value of name
-     * @param name the new value of name
-     */
-    public void setName(String name) {
+        if(!Validate.name(name, "^[A-Z][a-z]*$"))
+            throw new InvalidNameException();
+
         this.name = name;
     }
 
-    /**
-     * Get the value of name
-     * @return the value of name
-     */
-    public String getName() {
+    public String getName()
+    {
         return name;
     }
 
-    /**
-     * Set the value of designation
-     * @param designation the new value of designation
-     */
-    public void setDesignation(String designation) {
-        this.designation = designation;
+    public void setLastName(String lastName)
+    {
+        if(!Validate.name(lastName, "^[A-Z][a-z]*$"))
+            throw new InvalidNameException();
+
+        this.lastName = lastName;
+    }
+
+    public String getLastName()
+    {
+        return lastName;
+    }
+
+    public void setSurname(String surname)
+    {
+        if(!Validate.name(surname, "^[A-Z][a-z]*$"))
+            throw new InvalidNameException();
+
+        this.surname = surname;
+    }
+
+    public String getSurname()
+    {
+        return surname;
+    }
+
+    public void setNickname(String nickname)
+    {
+        if(!Validate.name(nickname, "^[a-z0-9]+$"))
+            throw new InvalidNameException();
+
+        this.nickname = nickname;
     }
 
-    /**
-     * Get the value of designation
-     * @return the value of designation
-     */
-    public String getDesignation() {
-        return designation;
+    public String getNickname()
+    {
+        return nickname;
+    }
+
+    public void setEmail(String email)
+    {
+        if(!email.matches("^(.+)@(.+)$"))
```

Comment:
@tatevikb is that correct regex for email ?

Reply:
"^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$" this one is correct regexp, I just picked up the simplest one for easy testing.